src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentDesignTimeNodeWriter.cs

```diff
@@ -650,10 +650,11 @@ private void WriteComponentAttributeInnards(CodeRenderingContext context, Compon
                 }
             }
 
-            bool NeedsTypeCheck(ComponentAttributeIntermediateNode n)
+            static bool NeedsTypeCheck(ComponentAttributeIntermediateNode n)
             {
-                return n.BoundAttribute != null && !n.BoundAttribute.IsWeaklyTyped();
-            }   
+                // Weakly typed attributes will have their TypeName set to null.
```

Comment:
Because when the attribute under consideration is `@onclick:preventDefault`, the value of `BoundAttribute` property will be the BoundAttributeDescriptor associated with `@onclick`. This means `n.BoundAttribute.IsWeaklyTyped()` will return `true` which in turn means we won't get type checking for this. But we want type checking for our attribute.